CRM/Upgrade/Incremental/php/FiveThirtyFour.php

```diff
@@ -153,10 +153,11 @@ public static function migrateSmtpPasswords(CRM_Queue_TaskContext $ctx) {
       $value = unserialize($setting['value']);
       $plain = CRM_Utils_Crypt::decrypt($value['smtpPassword']);
       $value['smtpPassword'] = $cryptoToken->encrypt($plain, 'CRED');
-      CRM_Core_DAO::executeQuery('UPDATE civicrm_setting SET value = %2 WHERE id = %1', [
-        1 => [$setting['id'], 'Positive'],
-        2 => [serialize($value), 'String'],
-      ]);
+      $dao = new CRM_Core_BAO_Setting();
+      $dao->id = $setting['id'];
+      $dao->find(TRUE);
+      $dao->value = serialize($value);
+      $dao->save();
```

Comment:
@totten doing $dao->value does not do any escaping AFAIK which is the difference here. all it will do is put quotes around the value to be set based on https://github.com/civicrm/civicrm-packages/blob/master/DB/DataObject.php#L1365

Reply:
I'm fairly certain many things would be broken if `$dao->value` didn't escape.

You can see escaping empirically with [this script](https://gist.github.com/totten/311b1308d6ae6f8f7d2785bbb55d4460) which writes a setting using different mechanisms:

```
[bknix-min:~/bknix/build/dmaster/web/sites/all/modules/civicrm] cv scr settingValueExamples.php 
Array
(
    [0] => Array
        (
            [id] => 6
            [value] => s:31:"the 'example' for "setting svc"";
        )

)
Array
(
    [0] => Array
        (
            [id] => 6
            [value] => s:32:"the 'example' for "executeQuery"";
        )

)
Array
(
    [0] => Array
        (
            [id] => 6
            [value] => s:23:"the 'example' for "DAO"";
        )

)
```

If any of those failed to escape, then at least one of those would examples fail to write. Instead, they all succeed in writing - and they read back as expected.